oml/lightning/callbacks/metric.py

```diff
@@ -54,11 +55,14 @@ def __init__(
         self._collected_samples = 0
         self._ready_to_accumulate = False
 
-    def _calc_expected_samples(self, trainer: pl.Trainer, dataloader_idx: int) -> int:
-        return self.samples_in_getitem * len(trainer.val_dataloaders[dataloader_idx].dataset)
+    def _calc_expected_samples(self, trainer: pl.Trainer, dataloader_idx: int = 0) -> int:
```

Comment:
any reason to default values for dataloader_idx? The same for other functions

Reply:
it fails otherwise, so they must be here